zerver/tests/test_bugdown.py

```diff
@@ -832,6 +832,14 @@ def test_links_in_topic_name(self) -> None:
         converted_topic = bugdown.topic_links(realm.id, msg.topic_name())
         self.assertEqual(converted_topic, [u'https://google.com/hello-world'])
 
+        msg.set_topic_name("Without scheme google.com/hello-world")
+        converted_topic = bugdown.topic_links(realm.id, msg.topic_name())
+        self.assertEqual(converted_topic, [u'https://google.com/hello-world'])
+
+        msg.set_topic_name("Without scheme random.words/hello-world")
+        converted_topic = bugdown.topic_links(realm.id, msg.topic_name())
+        self.assertEqual(converted_topic, [])
```

Comment:
We should have a test for an `http://` URL; I think this code might break those.

Reply:
Actually, the code is fine; but it does need a test.

Reply:
It *does* have an http test in the testcase below this one:

`msg.set_topic_name("Try out http://ftp.debian.org, https://google.com/ and https://google.in/.")`